src/shims/env.rs

```diff
@@ -111,4 +112,33 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             Ok(-1)
         }
     }
+
+    fn getcwd(
+        &mut self,
+        buf_op: OpTy<'tcx, Tag>,
+        size_op: OpTy<'tcx, Tag>,
+    ) -> InterpResult<'tcx, Scalar<Tag>> {
+        let this = self.eval_context_mut();
+
+        if this.machine.communicate {
```

Comment:
make this an early return by inverting the condition and throwing unsup instead of increasing the nesting level